swa-price-drop.js

```diff
@@ -321,17 +340,165 @@ const checkSouthwest = (flightConfig, cb) => {
+                for (let rawPrice of prices) {
+                  const priceMatch = rawPrice.toString().match(/\$.*?(\d+)/);
+                  const price = parseInt(priceMatch[1]);
+                  logger2.debug("Found price " + price + " for return flight " + match + " on " + flightConfig.returnDate);
+                  returnFares.push({'Flight':flightNumber, 'price':price});
+                }
+              }
+            }
+          })
+        ]
+      })
+      .done(() => {
+        // sort flights by price, log, and compare configured flights to searched flights
+        outboundFares.sort(function(a,b){
+          return a.price-b.price;
+        })
+        returnFares.sort(function(a,b){
+          return a.price-b.price;
+        })
+        const outboundPrice = parseInt(flightConfig.outboundPrice);
+        const returnPrice = parseInt(flightConfig.returnPrice);
+        const lowestOutboundFare = outboundFares[0].price;
+        const lowestReturnFare = returnFares[0].price;
+
+        logger2.debug('Lowest outbound price for flights is ' + lowestOutboundFare + ' on flight number #' + outboundFares[0].Flight);
+        logger2.debug('Notification threshold outbound price for flights is ' + outboundPrice);
+        logger2.debug('Lowest return price for flights is ' + lowestReturnFare + ' on flight number #' + returnFares[0].Flight);
+        logger2.debug('Notification threshold return price for flights is ' + returnPrice);
+
+        if (lowestOutboundFare < outboundPrice) {
+          const message = "Price Drop: Outbound flight #" + flightConfig.outboundFlightNumber + " " +
+            flightConfig.originAirport + "->" + flightConfig.destinationAirport +
+            " on " + flightConfig.outboundDate +
+            " is now $" + lowestOutboundFare + " (was $" + flightConfig.outboundPrice + ")";
+          notify(message);
+
+          // update fare price in config
+          flightConfig.outboundPrice = lowestOutboundFare;
+          // flag to indicate we need to re-write yaml config
+          rewriteYamlConfig = true;
+        }
+
+        if (lowestReturnFare < returnPrice) {
+          const message = "Price Drop: Return flight #" + flightConfig.returnFlightNumber + " " +
+            flightConfig.destinationAirport + "->" + flightConfig.originAirport +
+            " on " + flightConfig.returnDate +
+            " is now $" + lowestReturnFare + " (was $" + flightConfig.returnPrice + ")";
+          notify(message);
+
+          // update fare price in config
+          flightConfig.returnPrice = lowestReturnFare;
+          // flag to indicate we need to re-write yaml config
+          rewriteYamlConfig = true;
+        }
+
+        updatedFlights.push(flightConfig);
+        cb();
+      })
+  }
+
+}
+
+
+
 // main
 async.series([
   function(next) {
     async.each(config.flights, function(flight, cb) {
-      checkSouthwest(flight, cb);
+      checkSouthwestDaily(flight, cb);
```

Comment:
Does this mean that 'checkSouthwest' is deprecated or is it called elseware?

The two methods seem to share a fair bit of code, if 'checkSouthwest' isn't deprecated you may want to abstract. You may also want to change your CSS identifiers to static variables set initially to make it easier to change them if SW changes their site CSS. 

I don't know how much of that is possible in JS